core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -276,6 +277,33 @@ class DiplomacyManager() : IsPartOfGameInfoSerialization {
         }
     }
 
+    private fun believesSameReligion(): Boolean? {
+        // believing same/different religion - modifiers
+        // what is the majority religion of civInfo?
+        val civMajorityReligion = civInfo.gameInfo.religions.values.firstOrNull {
+            civInfo.religionManager.isMajorityReligionForCiv(it)
```

Comment:
Do we not have a function for "get majority religion" in religion manager?

Reply:
I'll check.
Also, should I rather implement such in there (religion manager), if we don't?

Reply:
We do, only City-wise in the CityReligionManager.

Reply:
> "but"...
> ```kotlin
>     fun getMajorityReligion(): Religion? {
>         val counter = Counter<Religion>()
>         for (city in civInfo.cities)
>             city.religion.getMajorityReligion()?.let { counter.add(it, 1) }
>         return counter.maxByOrNull { it.value }?.takeIf { it.value > civInfo.cities.size / 2 }?.key
>     }
> ```
> in ReligionManager would likely be quite a bit more performant: 1 instead of N cities-loops.

Thank you 🙂, I will use this instead. I had looked for such a function but had not found it.

Reply:
> Mind you - I tend to write a little too "kotlin-y" on quick&dirty stuff, and think about readability later. Resolving `?.let` and/or `?.takeIf` to step-by-step verbose code might be easier on the eye - I'll trust you with that task.

I hope your trust was well placed 🙂
You can have a look to the new version of the code.
Thank you very much again in advance.